contracts/Zauction.sol

```diff
@@ -93,33 +97,103 @@ contract ZAuction is Initializable, OwnableUpgradeable {
     }
     require(!consumed[bidder][auctionId], "zAuction: data already consumed");
 
-    // Will truncate any decimals
-    uint256 royalty = bid / 10;
+    uint256 rootId = rootDomainId[tokenId];
+    if (rootId == 0) {
+      rootId = rootDomainIdOf(tokenId);
+      rootDomainId[tokenId] = rootId;
+    }
 
-    IERC721 nftContract = IERC721(nftAddress);
     consumed[bidder][auctionId] = true;
-    // Bidder -> Owner, send funds
-    SafeERC20.safeTransferFrom(token, bidder, msg.sender, bid - royalty);
-    // Bidder -> Registrar, pay royalty
-    SafeERC20.safeTransferFrom(
-      token,
-      bidder,
-      registrar.minterOf(tokenId),
-      royalty
-    );
+
+    // Transfer payment and royalties
+    paymentTransfers(bidder, bid, msg.sender, rootId, tokenId);
+
     // Owner -> Bidder, send NFT
-    nftContract.safeTransferFrom(msg.sender, bidder, tokenId);
+    registrar.safeTransferFrom(msg.sender, bidder, tokenId);
+
     emit BidAccepted(
       auctionId,
       bidder,
       msg.sender,
       bid,
-      address(nftContract),
+      address(registrar),
       tokenId,
       expireBlock
     );
   }
 
+  function paymentTransfers(
```

Comment:
Organize functions in smart contracts in this order
`public/external`
`internal`
`private`